mlir/lib/Dialect/MemRef/Utils/MemRefUtils.cpp

```diff
@@ -81,11 +81,10 @@ std::pair<LinearizedMemRefInfo, OpFoldResult> getLinearizedMemRefOffsetAndSize(
 
   // Adjust linearizedIndices and size by the scale factor (dstBits / srcBits).
   int64_t scaler = dstBits / srcBits;
-  addMulMap = addMulMap.floorDiv(scaler);
   mulMap = mulMap.floorDiv(scaler);
 
   OpFoldResult linearizedIndices = affine::makeComposedFoldedAffineApply(
-      builder, loc, addMulMap, offsetValues);
+      builder, loc, addMulMap.floorDiv(scaler), offsetValues);
```

Comment:
I don't follow. I think it is needed because the linearized indices needs `addMulMap.floorDiv(...)` and the intraVectorOffset needs `addMulMap % scaler`?

Reply:
I think this small change needed